src/AutoMapper/TypeDetails.cs

```diff
@@ -99,9 +99,9 @@ public struct DestinationMemberName
 
         public IEnumerable<DestinationMemberName> DestinationMemberNames { get; set; }
 
-        private IEnumerable<MethodInfo> BuildPublicNoArgExtensionMethods(IEnumerable<MethodInfo> sourceExtensionMethodSearch)
+        private IEnumerable<MethodInfo> BuildPublicNoArgExtensionMethods(IEnumerable<MethodInfo> sourceExtensionMethodSearch, Func<MethodInfo, bool> shouldMapMethod)
         {
-            var explicitExtensionMethods = sourceExtensionMethodSearch.Where(method => method.GetParameters()[0].ParameterType == Type);
+            var explicitExtensionMethods = sourceExtensionMethodSearch.Where(shouldMapMethod).Where(method => method.GetParameters()[0].ParameterType == Type);
```

Comment:
I think we can remove all the filters inside the method and simply filter the input parameter when calling the method.

Reply:
Make sense!